src/main/java/nl/novi/serviceconnect/services/TransactionService.java

```diff
@@ -0,0 +1,118 @@
+package nl.novi.serviceconnect.services;
+
+import nl.novi.serviceconnect.dtos.TransactionInputDto;
+import nl.novi.serviceconnect.dtos.TransactionOutputDto;
+import nl.novi.serviceconnect.exceptions.RecordNotFoundException;
+import nl.novi.serviceconnect.helpper.Helpers;
+import nl.novi.serviceconnect.models.ServiceRequest;
+import nl.novi.serviceconnect.models.Transaction;
+import nl.novi.serviceconnect.repository.ServiceRequestRepository;
+import nl.novi.serviceconnect.repository.TransactionRepository;
+import org.springframework.stereotype.Service;
+import nl.novi.serviceconnect.helpper.Mapper;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TransactionService implements ITransactionService {
+    private final TransactionRepository transactionRepository;
+
+    private final ServiceRequestRepository requestRepository;
+    public TransactionService(TransactionRepository repository, ServiceRequestRepository requestRepository) {
+        this.transactionRepository = repository;
+        this.requestRepository = requestRepository;
+    }
+
+    private String saveFile() {
+        try {
+            File targetDirectory = new File("C:\\Java-HBO\\LesOpdrachten\\EindProject\\ServiceConnect\\pdfs");
+
+            if (!targetDirectory.exists()) {
+                targetDirectory.mkdirs();
+            }
+
+            String fileName = "invoice_" + System.currentTimeMillis() + ".pdf";
+
+            String filePath = targetDirectory.getAbsolutePath() + File.separator + fileName;
+
+            return filePath;
+        } catch (Exception e) {
+
+            throw new RuntimeException("Error generating and saving PDF file", e);
+        }
+    }
+
+    @Override
+    public TransactionOutputDto createTransaction(TransactionInputDto transactionInputDto) {
+
+        Transaction transaction = Mapper.fromDtoToTransaction(transactionInputDto);
+
+        String filePath = saveFile();
+        transaction.setInvoice(new File(filePath));
+
+        ServiceRequest serviceRequest = requestRepository.findById(transactionInputDto.getServiceRequestId())
+                .orElseThrow(() -> new RecordNotFoundException("ServiceRequestId not found with id: " + transactionInputDto.getServiceRequestId()));
```

Comment:
Dit zou je dus aan het begin van de methode kunnen checken om te voorkomen dat je een bestand aanmaakt terwijl dat helemaal niet nodig is / kan.